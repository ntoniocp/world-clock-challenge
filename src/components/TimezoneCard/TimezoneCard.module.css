.card {
  background-color: #c7def0;
  border: 1px solid black;
  border-radius: 5px;
  font-size: 14px;
  min-width: 130px;
  width: 100%;
  padding: 18px;
  position: relative;
  text-align: center;
}

.card_title {
  font-weight: bold;
  font-size: inherit;
  margin-bottom: 2px;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.close-icon {
  cursor: pointer;
  position: absolute;
  right: 8px;
  top: 4px;
  user-select: none;
}

.close-icon:hover {
  font-weight: bold;
}
