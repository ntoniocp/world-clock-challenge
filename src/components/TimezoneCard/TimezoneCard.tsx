import styles from './TimezoneCard.module.css'

interface Props {
  name: string,
  date: string,
  time: string
  onClose: Function
}

function TimezoneCard({name, date, time, onClose}: Props) {
  return (
    <div className={styles.card}>
      <span className={styles['close-icon']} onClick={() => onClose()}>X</span>
      <h2 className={styles['card_title']}>{name}</h2>
      <p className={styles.date}>{date}</p>
      <p className={styles.time}>{time}</p>
    </div>
  )
}

export default TimezoneCard

