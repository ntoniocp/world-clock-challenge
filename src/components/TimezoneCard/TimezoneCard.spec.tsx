import TimezoneCard from './TimezoneCard'
import renderer from 'react-test-renderer';

const cardProps = {
  name: 'Test/Timezone',
  date: '14/12/2021',
  time: '03:42PM',
  onClose: () => {}
}

describe('TimezoneCard Component', () => {
  test('It matches the snapshot', () => {
    const tree = renderer.create(<TimezoneCard {...cardProps} />).toJSON();
    expect(tree).toMatchSnapshot()
  })
})