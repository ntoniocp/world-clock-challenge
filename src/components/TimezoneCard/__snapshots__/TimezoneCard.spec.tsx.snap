// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimezoneCard Component It matches the snapshot 1`] = `
<div
  className="card"
>
  <span
    className="close-icon"
    onClick={[Function]}
  >
    X
  </span>
  <h2
    className="card_title"
  >
    Test/Timezone
  </h2>
  <p
    className="date"
  >
    14/12/2021
  </p>
  <p
    className="time"
  >
    03:42PM
  </p>
</div>
`;
